shielddb-core/src/main/java/nf/fr/k49/shielddb/core/ShieldDBBuilder.java

```diff
@@ -21,14 +22,22 @@ public ShieldDBBuilder() {
 	public List<T> build() {
 		final BottomShield<T> bottom = new BottomShield<>(mapper, storage);
 		shields.add(bottom);
-		for(int i = 0; i < shields.size()-1 ; i++) {
-			final ShieldDBShield<T> cur = shields.get(i);
-			final ShieldDBShield<T> next = shields.get(i+1);
-			cur.setNextShield(next);
-		}
+		doNestShields();
 		return shields.get(0);
 	}
 
+	private void doNestShields() {
+		Iterator<ShieldDBShield<T>> shieldIterator = shields.iterator();
+
+		ShieldDBShield<T> bindTo = shieldIterator.next();
+		while (shieldIterator.hasNext()) {
+			ShieldDBShield<T> current = shieldIterator.next();
+			bindTo.setNextShield(current);
+			bindTo = current;
+		}
+	}
```

Comment:
By creating the doNestShields method, I think you split code that should stay in the build() method. Can you replace the doNestShields call by this method content and remove it?

Reply:
oookay. Why do you decide against extraction? For me, self describing code (in good named methods) a likely better than placing code in one method. 
Or is the build-pattern a reason for you?

Reply:
By the way: 
I changed the alg in using iterator. Is that fine for you or should I use the old alg?